client/webserver/site/src/js/order.ts

```diff
@@ -133,6 +137,212 @@ export default class OrderPage extends BasePage {
+      return
+    }
+
+    const tmpl = Doc.parseTemplate(matchCard)
+    tmpl.status.textContent = OrderUtil.matchStatusString(m)
+
+    const setCoin = (pendingName: string, linkName: string, coin: Coin) => {
+      const coinLink = tmpl[linkName]
+      const pendingSpan = tmpl[pendingName]
+      if (!coin) {
+        Doc.show(tmpl[pendingName])
+        Doc.hide(tmpl[linkName])
+        return
+      }
+      coinLink.textContent = coin.stringID
+      coinLink.dataset.explorerCoin = coin.stringID
+      setCoinHref(coinLink)
+      Doc.hide(pendingSpan)
+      Doc.show(coinLink)
+    }
+
+    setCoin('makerSwapPending', 'makerSwapCoin', makerSwapCoin(m))
+    setCoin('takerSwapPending', 'takerSwapCoin', takerSwapCoin(m))
+    setCoin('makerRedeemPending', 'makerRedeemCoin', makerRedeemCoin(m))
+    setCoin('takerRedeemPending', 'takerRedeemCoin', takerRedeemCoin(m))
+    setCoin('refundPending', 'refundCoin', m.refund)
+
+    if (m.status === OrderUtil.MakerSwapCast && !m.revoked && !m.refund) {
+      const c = makerSwapCoin(m)
+      tmpl.makerSwapMsg.textContent = confirmationString(c)
+      Doc.hide(tmpl.takerSwapMsg, tmpl.makerRedeemMsg, tmpl.takerRedeemMsg)
+      Doc.show(tmpl.makerSwapMsg)
+    } else if (m.status === OrderUtil.TakerSwapCast && !m.revoked && !m.refund) {
+      const c = takerSwapCoin(m)
+      tmpl.takerSwapMsg.textContent = confirmationString(c)
+      Doc.hide(tmpl.makerSwapMsg, tmpl.makerRedeemMsg, tmpl.takerRedeemMsg)
+      Doc.show(tmpl.takerSwapMsg)
+    } else if (inConfirmingMakerRedeem(m) && !m.revoked && !m.refund) {
+      tmpl.makerRedeemMsg.textContent = confirmationString(m.redeem)
+      Doc.hide(tmpl.makerSwapMsg, tmpl.takerSwapMsg, tmpl.takerRedeemMsg)
+      Doc.show(tmpl.makerRedeemMsg)
+    } else if (inConfirmingTakerRedeem(m) && !m.revoked && !m.refund) {
+      tmpl.takerRedeemMsg.textContent = confirmationString(m.redeem)
+      Doc.hide(tmpl.makerSwapMsg, tmpl.takerSwapMsg, tmpl.makerRedeemMsg)
+      Doc.show(tmpl.takerRedeemMsg)
+    } else {
+      Doc.hide(tmpl.makerSwapMsg, tmpl.takerSwapMsg, tmpl.makerRedeemMsg, tmpl.takerRedeemMsg)
+    }
+
+    const showMakerSwap = !m.isCancel && (makerSwapCoin(m) || !m.revoked)
+    const showTakerSwap = !m.isCancel && (takerSwapCoin(m) || !m.revoked)
+    const showMakerRedeem = !m.isCancel && (makerRedeemCoin(m) || !m.revoked)
+    const showTakerRedeem = !m.isCancel && (m.side !== OrderUtil.Maker) && (takerRedeemCoin(m) || !m.revoked)
+    const showRefund = !m.isCancel && (m.refund || (m.revoked && m.active))
+
+    if (showMakerSwap) Doc.show(tmpl.makerSwap)
+    else Doc.hide(tmpl.makerSwap)
+    if (showTakerSwap) Doc.show(tmpl.takerSwap)
+    else Doc.hide(tmpl.takerSwap)
+    if (showMakerRedeem) Doc.show(tmpl.makerRedeem)
+    else Doc.hide(tmpl.makerRedeem)
+    if (showTakerRedeem) Doc.show(tmpl.takerRedeem)
+    else Doc.hide(tmpl.takerRedeem)
+    if (showRefund) Doc.show(tmpl.refund)
+    else Doc.hide(tmpl.refund)
```

Comment:
Some of the existing code here is clearly from very early development. There is also `Doc.setVis` now, so you could do 1 or 2 lines per element, e.g.

```Javascript
Doc.setVis(!m.isCancel && (makerSwapCoin(m) || !m.revoked), tmpl.makerSwap)
```